File: Python/Collection/Async.py
import asyncio

# Hauptfunktion, die asynchrone Aufgaben ausführt
async def run_async():
    print("Start der asynchronen Methoden")

    # Zwei Aufgaben parallel starten
    task1 = asyncio.create_task(task1_async())
    task2 = asyncio.create_task(task2_async())

    # Warten, bis beide Aufgaben fertig sind
    await asyncio.gather(task1, task2)

    print("Ende der asynchronen Methoden")

# Erste asynchrone Aufgabe
async def task1_async():
    print("Task1 gestartet")
    await asyncio.sleep(2)  # 2 Sekunden warten
    print("Task1 beendet")

# Zweite asynchrone Aufgabe
async def task2_async():
    print("Task2 gestartet")
    await asyncio.sleep(3)  # 3 Sekunden warten
    print("Task2 beendet")

# Die asynchrone Hauptfunktion ausführen
asyncio.run(run_async())

File: C#/Async.cs
using System;
using System.Threading.Tasks;

public class Async
{
    public Async()
    {
        RunAsync().Wait(); //Ruft die asynchrone Methode synchron auf, um sie im Konstruktor auszuführen
    }

    public async Task RunAsync()
    {
        Console.WriteLine("Start der asynchronen Methoden");

        //Starte zwei Aufgaben parallel
        Task task1 = Task1Async();
        Task task2 = Task2Async();

        //Warten, bis beide Aufgaben abgeschlossen sind
        await Task.WhenAll(task1, task2);

        Console.WriteLine("Ende der asynchronen Methoden");
    }

    private async Task Task1Async()
    {
        Console.WriteLine("Task1 gestartet");
        await Task.Delay(2000); //Simuliert eine Verzögerung von 2 Sekunden
        Console.WriteLine("Task1 beendet");
    }

    private async Task Task2Async()
    {
        Console.WriteLine("Task2 gestartet");
        await Task.Delay(3000); //Simuliert eine Verzögerung von 3 Sekunden
        Console.WriteLine("Task2 beendet");
    }
}
